src/app/screens/LNURLRedeem/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { CaretLeftIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
+import Button from "@components/Button";
+import Container from "@components/Container";
+import Header from "@components/Header";
+import IconButton from "@components/IconButton";
+import TextField from "@components/form/TextField";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useLocation, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import QrcodeAdornment from "~/app/components/QrcodeAdornment";
+import lnurlLib from "~/common/lib/lnurl";
+import { isLNURLDetailsError } from "~/common/utils/typeHelpers";
+
+function LNURLRedeem() {
+  const { t } = useTranslation("translation", { keyPrefix: "lnurlredeem" });
+  const location = useLocation();
+
+  const [lnurlwithdraw, setlnurlwithdraw] = useState(
```

Comment:
lnurlwithdraw seems like the wrong variable name here. Should this be `lnurl` or something?

Reply:
setter should have a capital letter after `set`

Reply:
Can you add a comment explaining why location.state is used? (comes from QRScanner screen). I wonder if `decodedQR` is better than `decodedText`?

Reply:
but the expected text from the form is lnurlWithrdraw string right? we are just validating after that if its lnurl and has tag of withdraw request or not. WDYT?

Reply:
I think "LNURL-withdraw" is the name of the protocol, that's why I suggested call it just "lnurl", maybe "lnurlWithdrawLink" could also be good?

For `decodedText`, this is not explanatory at all where it comes from. That's why I suggested `decodedQR`

Reply:
changed name from lnurlwithdraw to lnurlWithdrawLink